### PR TITLE
Share list with existing users

### PR DESCRIPTION
**Pull request by aellispierce**

### CODE REVIEW

File: spec/features/user_invites_someone_to_share_list_spec.rb [hunk 1]
```diff
@@ -20,6 +20,26 @@
     end
   end
 
+  scenario "who does have an account" do
+    user = create(:user, email: "sarah@amazing.com")
```
Comment (drapergeek) [inline spec/features/user_invites_someone_to_share_list_spec.rb]: How about: `email: "user-to-share-with@example.com"`

File: spec/features/user_invites_someone_to_share_list_spec.rb [hunk 2]
```diff
@@ -20,6 +20,26 @@
...
+    visit list_path(list, as: user)
+
+    invite_user "sarah@amazing.com"
+
+    using_session :sarah do
+      open_last_email
+      click_first_link_in_email
+      fill_in "Email", with: "sarah@amazing.com"
+      fill_in "Password", with: "password"
+      click_on "Accept Invitation"
```
Comment (drapergeek) [inline spec/features/user_invites_someone_to_share_list_spec.rb]: Add a new line between the action and verification.

File: spec/features/user_invites_someone_to_share_list_spec.rb [hunk 3]
```diff
@@ -20,6 +20,23 @@
...
+  scenario "who does have an account" do
+    user = create(:user)
+    invited_user = create(:user, email: "user_to_share_with@example.com")
+    list = create(:list, user: user, name: "Bob's List")
+
+    visit list_path(list, as: user)
+
+    invite_user "user_to_share_with@example.com"
+
+    using_session :user_to_share_with do
```
Comment (drapergeek) [inline spec/features/user_invites_someone_to_share_list_spec.rb:32]: :heart:

File: app/controllers/invitations_controller.rb [hunk 4]
```diff
@@ -1,16 +1,27 @@
 class InvitationsController < ApplicationController
   def create
-    deliver_invitation
+    if user_exists?
+      share_list
+    else
+      deliver_invitation
+    end
```
Comment (drapergeek) [inline app/controllers/invitations_controller.rb:7]: Mind adding a new line after the conditional here?

File: app/controllers/invitations_controller.rb [hunk 5]
```diff
@@ -1,16 +1,27 @@
...
+  attr_reader :user
+
   def deliver_invitation
     @invitation = Invitation.new(invitation_params)
     @invitation.deliver
   end
 
+  def share_list
+    shared_list = ListShare.create(list: list, user: user)
+    @list_page = shared_list.list
```
Comment (drapergeek) [inline app/controllers/invitations_controller.rb]: How about calling this the `redirect_location`?

File: app/controllers/invitations_controller.rb [hunk 6]
```diff
@@ -19,10 +30,18 @@ def invitation_params
...
   def redirect_with_notice
-    redirect_to @invitation.list, notice: "Invitation Sent"
+    redirect_to list_page, notice: "Invitation Sent"
   end
 
   def list
     @list ||= current_user.lists.find(params[:list_id])
   end
+
+  def list_page
```
Comment (drapergeek) [inline app/controllers/invitations_controller.rb]: How about calling this the `redirect_location`?

File: app/models/acceptance.rb [hunk 7]
```diff
@@ -37,6 +37,6 @@ def unused_invitation
   end
 
   def build_user
-    User.new(user_attributes)
+    User.find_by(email: user_attributes[:email]) || User.new(user_attributes)
```
Comment (drapergeek) [inline app/models/acceptance.rb]: Can this be removed now?

**Comment by drapergeek** [comment]
LGTM after comments

File: app/models/acceptance.rb (hunk 7 above)
Comment (aellispierce) [inline app/models/acceptance.rb]: oh yep

**Pull request merged**